crates/symbolicator/Cargo.toml

```diff
@@ -10,6 +10,7 @@ anyhow = "1.0.57"
 apple-crash-report-parser = "0.4.2"
 async-trait = "0.1.53"
 axum = { version = "0.5.4", features = ["multipart"] }
+axum-server = { version = "0.4.0", features = [ "tls-rustls"] }
```

Comment:
would it be worth putting this behind a cargo feature?

/cc @Swatinem

Reply:
That might be a good idea, yes. Outgoing connections use native-tls (aka openssl) for now.

Reply:
So you want me to make the https server optional then? I'll have to research how to do that; first I'll try moving its creation into a separate function (and for symmetry, do the same thing for the HTTP server). Any suggestions about how to go about this feature business are appreciated. :-)

Reply:
There is 3 parts to that:
1. Mark the `axum-server` dependency as `optional: true`
2. Introduce a `[features]` section with at least a `https = ["axum-server"]` entry. If you want to make this opt-out instead of opt-in by default, you can set `default = ["https"]` as well.
3. In code, you can sprinkle `#[cfg(feature = "https")]` all over the place, for example above code blocks, which are then conditionally compiled.

Reply:
Thank you for the extra detail, I'll fix this.

Reply:
I just realized that if the axum-server dependency is optional, the HTTP server will necessarily need to continue to use axum. This will lead to two different web server frameworks to be used, with different APIs and abilities. Not sure if that's a good thing...

Reply:
Ah absolutely, I missed that. In that case, you can have `https = ["axum-server/tls-rustls"]` as the feature, which will activate that specific feature of an existing dependency.

Reply:
Thanks, I'll look into fixing this.

Reply:
Just pushed a change to make the https server an optional feature.

Reply:
In that case you don’t need the `features` here, as that feature is activated by the feature flag down below.